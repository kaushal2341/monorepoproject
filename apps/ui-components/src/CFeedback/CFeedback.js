import React from 'react';
import {FormControl} from "react-bootstrap";
import Proptypes from 'prop-types';

const CFeedback = props => {
    const {
        as,
        className,
        message,
        id,
        type
    } = props;
    return (
        <FormControl.Feedback
            as={as}
            className={className}
            id={"feedback_".concat(id)}
            type={type}
        >
            {message}
        </FormControl.Feedback>
    )
};

CFeedback.propTypes = {
    as: Proptypes.elementType,
    id: Proptypes.string.isRequired,
    message: Proptypes.string.isRequired,
    type: Proptypes.string
};
export default CFeedback;