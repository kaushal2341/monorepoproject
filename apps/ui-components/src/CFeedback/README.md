# CFeedback

Component that shows feedback on FormControls(form validation message) made using react-bootstrap's \
`FormControl.Feedback` component.

To use it ,first import the component,

```jsx harmony
import {CFeedback} from '@cogent/ui-components';
```
Then use it as, 

```jsx harmony
<CForm id="test"  validated={true}>
  <CFormGroup
    id="email">
    <CFLabel
       id="name"
       labelName="Email"
    />
    <CFControl
       id="email"
       type="email"
       name="email"
       placeholder="Enter your email"
       size="lg"
       required={true}
       onChange={(e) => console.log(e.target.value)}
    />
    <CFeedback id="email" message="Email looks good."/>
    <CFeedback id="email" message="Email looks bad." type="invalid"/>
  </CFormGroup>
</CForm>
```
``Note:`` To use CFeedback, we have to use it within the CForm component with validated props set to true.
## CFeedback props

```text
as: elementType - You can use a custom element type for this component.

className: string - Css classname for additional styling.

id: string - Id of the component.

message: string - A feedback message to display.

type: 'valid' | 'invalid' -  defaultValue:'valid' - Specify whether the feedback is for valid or invalid fields.                                                  
```