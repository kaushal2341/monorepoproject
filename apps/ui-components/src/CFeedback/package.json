{
  "name": "@cogent/c-feedback",
  "version": "0.0.0",
  "main": "CFeedback.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
