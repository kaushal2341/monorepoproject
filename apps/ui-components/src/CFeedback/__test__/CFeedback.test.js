import CFeedback from "../CFeedback";

expect.addSnapshotSerializer(enzymeSerializer);

describe('CFeedback Component Tests', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CFeedback id="test" message="You're doing good!"/>);
    });

    it('should be defined', () => {
        expect(wrapper.find('CFeedback')).toBeDefined();
    });

    it('should have only one Form.Control component', () => {
        expect(wrapper.find('#feedback_test').length).toBe(1);
    });

    it('should have a feedback message', () =>{
        expect(wrapper.find('#feedback_test').text()).not.toBe('');
    });

    it('should have all props available', () => {
        let propsAvailable = [
            'as',
            'className',
            'id',
            'type'
        ];
        let propsOfFormControl = Object.keys(wrapper.find('#feedback_test').props());
        propsAvailable.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfFormControl[i])
        ));
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
});