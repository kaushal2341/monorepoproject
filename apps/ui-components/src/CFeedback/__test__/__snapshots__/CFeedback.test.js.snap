// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFeedback Component Tests Snapshot Testing should match snapshot 1`] = `
<Feedback
  id="feedback_test"
  type="valid"
>
  You're doing good!
</Feedback>
`;
